src/test/java/com/thebois/models/world/WorldTests.java

```diff
@@ -437,4 +443,63 @@ public void findNearestIncompleteStructureFindsNoIncompleteStructure() {
         assertThat(foundStructure.isEmpty()).isTrue();
     }
 
+    @Test
+    public void tryCreateStructureOfTypeTownHallCreatesAStructureIfTownHallIsAlreadyPresent() {
+        // Arrange
+        final World world = new TestWorld(50);
+        world.tryCreateStructure(StructureType.TOWN_HALL, new Position(10, 10));
+        final Position createTownHallPosition = new Position(12, 12);
+
+        // Act
+        world.tryCreateStructure(StructureType.TOWN_HALL, createTownHallPosition);
+
+        // Assert
+        assertThat(world
+                       .getStructures()
+                       .stream()
+                       .anyMatch(structure -> structure
+                           .getType()
+                           .equals(StructureType.HOUSE))).isTrue();
+    }
```

Comment:
I've changed the logic in World so that a second townhall can't be placed and tryCreateStructure returns false. So the test now checks that it is not possible to create a second townhall. Thanks for spotting 👍